### ISSUE TITLE
Rocketry Expanded crashes when attempting to add a module between two existing modules

### ISSUE THREAD

**Issue by TroublesomeCorvid** [Sgt-Imalas/Sgt_Imalas-Oni-Mods#25]
I'm not quite sure it was actually Rocketry Expanded that caused the crash, even though the game said so.  I'm also using Ash's Improved Rocket Controls mod.  If it turns out that's the issue, let me know so I can report the bug to them.

I had a CO2 engine on a launch pad with an Extended Solo Spacefarer Nosecone - nothing else, just the engine and the nosecone.  I selected the engine, and clicked the "+" button with the intent of adding another module between the two.  That was when the game Oopsa-daisy'd on me.  The following is the report produced by the Oopsa-daisy, and attached it the Player.log file.

```
NullReferenceException

Rockets_TinyYetBig.CategorizedModMenuPatches+BuildableStatesCategoryPatch.Prefix (SelectModuleSideScreen __instance, System.Collections.Generic.Dictionary`2[BuildingDef,System.Boolean]& ___moduleBuildableState, BuildingDef ___selectedModuleDef) (at <d4decd678d7741a6b42acb7de8450a74>:0)
(wrapper dynamic-method) SelectModuleSideScreen.SelectModuleSideScreen.UpdateBuildableStates_Patch2(SelectModuleSideScreen,object)
SelectModuleSideScreen.SetTarget (UnityEngine.GameObject new_target) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
RocketModuleSideScreen.ClickAddNew (System.Single scrollViewPosition, BuildingDef autoSelectDef) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
RocketModuleSideScreen.<OnSpawn>b__18_0 () (at <a0b1e63c8d674793957c96bc63afc3da>:0)
KButton.SignalClick (KKeyCode btn) (at <3f146e4773e642efa0f9e57aaa8e84c1>:0)
KButton.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at <3f146e4773e642efa0f9e57aaa8e84c1>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <0b23117e1c7442f8b1842fe2f3bf6fcb>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <0b23117e1c7442f8b1842fe2f3bf6fcb>:0)
UnityEngine.EventSystems.EventSystem:Update()

Build: U43-531669-S
```

[Player.log](https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/files/10184132/Player.log)

**Comment by Sgt-Imalas** [comment]
I have a fix for that ready, I just cannot release that update on steam since it also contains half baked space station stuff.
If you are fine with that, you can use this beta release here: https://github.com/Sgt-Imalas/Sgt_Imalas-Oni-Mods/releases/tag/BETA_v2.9.9_RocketryExpanded

I highly recomment to avoid the affromentioned half baked space stations in that release, otherwise everything else should be working

**Comment by Sgt-Imalas** [comment]
fixed in release 2.4
